src/app/app.component.ts

```diff
@@ -513,16 +513,22 @@ export class AppComponent implements OnInit, AfterViewInit, CanvasTableSelectLis
     });
   }
 
-  async emptyTrash(trashFolderName: string) {
-    console.log('found trash folder with name', trashFolderName);
+  async emptyTrash(trashFolder: FolderListEntry) {
+    console.log('found trash folder with name', trashFolder.folderName);
     const messageLists = await this.rmmapi.listAllMessages(
       0, 0, 0,
       RunboxWebmailAPI.LIST_ALL_MESSAGES_CHUNK_SIZE,
-      true, trashFolderName
+      true, trashFolder.folderName
     ).toPromise();
-    await this.rmmapi.deleteMessages(messageLists.map(msg => msg.id)).toPromise();
+
+    // remove local copies
+    this.searchService.deleteMessages(messageLists.map((msg) => msg.id));
+    this.searchService.updateIndexWithNewChanges();
+    // empty remote folder
+    await this.rmmapi.emptyFolder(trashFolder.folderId);
```

Comment:
this is missing a `toPromise()` to be effective (`emptyFolder()` returns an Observable).

Reply:
Added, thanks!